crates/primitives-traits/src/transaction/signed.rs

```diff
@@ -184,45 +184,114 @@ where
 }
 
 #[cfg(feature = "op")]
-impl SignedTransaction for op_alloy_consensus::OpPooledTransaction {
-    fn tx_hash(&self) -> &TxHash {
-        match self {
-            Self::Legacy(tx) => tx.hash(),
-            Self::Eip2930(tx) => tx.hash(),
-            Self::Eip1559(tx) => tx.hash(),
-            Self::Eip7702(tx) => tx.hash(),
+mod op {
+    use super::*;
+    use op_alloy_consensus::{OpPooledTransaction, OpTxEnvelope};
+
+    const DEPOSIT_SIGNATURE: Signature = Signature::new(U256::ZERO, U256::ZERO, false);
+
+    impl SignedTransaction for OpPooledTransaction {
```

Comment:
One thing I’m wondering for long-term, do you see value in making `SignedTransaction` support optional behaviors (e.g. via an is_signed() or has_signature() trait method)? Would make it easier for consumers to reason about signature-less variants like Deposit.
Curious what others think.

Reply:
rather implementing cranelift's `PackedOption` for signature, so wherever needed, a zero signature can be converted to an option wherever needed with `PackedOption::expand` instead of having an option everywhere (including where not needed, like l1)
https://docs.wasmtime.dev/api/cranelift_entity/packed_option/struct.PackedOption.html

Reply:
yeah, this is planned, but we first want to remove the reth specific tx types and then reduce scope of the fn signature call

actually let me start investigating where we even need this